mat/dense.go

```diff
@@ -281,6 +281,10 @@ func (m *Dense) rawRowView(i int) []float64 {
 // Slice panics with ErrIndexOutOfRange if the slice is outside the capacity
 // of the receiver.
 func (m *Dense) Slice(i, k, j, l int) Matrix {
+	if i == k || j == l {
+		panic(ErrZeroLength)
+	}
+
 	mr, mc := m.Caps()
 	if i < 0 || mr <= i || j < 0 || mc <= j || k <= i || mr < k || l <= j || mc < l {
```

Comment:
The condition relating i/k and j/l can be changed to `k < i` and `l < j` since the equality case is handled above.

I'm wondering now about the order of checks though. @btracey What do you think about this? Is an out of range check a higher priority than a zero length? With the change in the para above, that would work.